tt_metal/impl/device/device.cpp

```diff
@@ -2329,6 +2332,40 @@ const string Device::build_kernel_target_path(JitBuildProcessorType t, int i, co
     return bs.get_target_out_path(kernel_name);
 }
 
+void Device::insert_persistent_program(const Program& program) {
+    TT_ASSERT(program.is_finalized(), "Program must be finalized before inserting into persistent programs");
+    uint32_t programmable_core_count = hal.get_programmable_core_type_count();
+    // Ugly. We create a copy of the current program after it has been finalized so that we have an owned copy with extended lifetime for now
```

Comment:
@dmakoviichuk-tt ptal. I feel like TT-NN cache is not ready for this

Reply:
Copying the program is almost definitely being removed. I don't think you should think about how this affects ttnn until the changes are in a more ready state. Ideally this has no effect on ttnn infra at all as a first pass.